azure-client-runtime/src/test/java/com/microsoft/azure/AzureProxyToRestProxyTests.java

```diff
@@ -597,10 +599,66 @@ public void AsyncHttpsHeadersRequest() {
         assertEquals("MyHeaderValue", headers.value("MyHeader"));
     }
 
+    @Host("https://httpbin.org")
+    private interface Service15 {
+        @GET("anything")
+        @ExpectedResponses({200})
+        Observable<HttpBinJSON> get();
+    }
+
+    @Test
+    public void service15Get() {
+        final Service15 service = createService(Service15.class);
+        try {
+            service.get();
+            fail("Expected exception.");
+        }
+        catch (InvalidReturnTypeException e) {
+            assertContains(e.getMessage(), "rx.Observable<com.microsoft.azure.HttpBinJSON>");
+            assertContains(e.getMessage(), "AzureProxyToRestProxyTests$Service15.get()");
+        }
+    }
+
+    @Host("http://httpbin.org")
```

Comment:
Oops, should've noticed you wrote this before I wrote a very similar thing.

Reply:
We were writing it at the same time. I'll merge your changes into this pull request.